android/app/src/main/java/org/fossasia/openevent/viewmodels/MainActivityViewModel.java

```diff
@@ -0,0 +1,226 @@
+import io.realm.RealmList;
+import io.realm.RealmResults;
+import timber.log.Timber;
+
+public class MainActivityViewModel extends ViewModel {
+    private RealmDataRepository realmRepo;
+    private LiveData<Event> eventLiveData;
+    private Event event;
+    private boolean newInstance = false;
+    private MutableLiveData<String> facebookPageNameLiveData;
+    private CompositeDisposable compositeDisposable = new CompositeDisposable();
+
+    public MainActivityViewModel() {
+        realmRepo = RealmDataRepository.getDefaultInstance();
+    }
+
+    public LiveData<Event> getEventSync() {
+        newInstance = false;
+        if (eventLiveData == null) {
+            event = realmRepo.getEventSync();
+            if (event == null) {
+                newInstance = true;
+                event = new Event();
+            }
+            LiveRealmDataObject<Event> liveRealmDataObject = RealmDataRepository.asLiveDataForObject(realmRepo.getEventSync());
+            eventLiveData = Transformations.map(liveRealmDataObject, input -> input);
+        }
+        return eventLiveData;
+    }
+
+    public MutableLiveData<String> downloadPageId(String pageId, String token) {
+        //Store the facebook page name in the shared preference from the database
+        if (facebookPageNameLiveData == null && !newInstance) {
+            facebookPageNameLiveData = new MutableLiveData<>();
+            RealmList<SocialLink> socialLinks = event.getSocialLinks();
+            if (socialLinks != null) {
+                RealmResults<SocialLink> facebookPage = socialLinks.where().equalTo("name", "Facebook").findAll();
+                if (facebookPage.size() == 0)
+                    return facebookPageNameLiveData;
+
+                SocialLink facebookLink = facebookPage.get(0);
+                String link = facebookLink.getLink();
+                String tempString = ".com";
+                String pageName = link.substring(link.indexOf(tempString) + tempString.length()).replace("/", "");
+
+                if (Utils.isEmpty(pageName))
+                    return facebookPageNameLiveData;
+
+                facebookPageNameLiveData.setValue(pageName);
+            }
+        }
+
+        if (pageId == null && facebookPageNameLiveData.getValue() != null) {
+                    Disposable facebookDisposable =APIClient.getFacebookGraphAPI().getPageId(facebookPageNameLiveData.getValue(),
+                    token)
+                    .subscribeOn(Schedulers.io())
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe(facebookPageId -> {
+                                String id = facebookPageId.getId();
+                                SharedPreferencesUtil.putString(ConstantStrings.FACEBOOK_PAGE_ID, id);
+                            },
+                            throwable -> Timber.d("Facebook page id download failed: " + throwable.toString()));
+            compositeDisposable.add(facebookDisposable);
+        }
```

Comment:
This whole block seems to be doing something else entirely, please move it to another function